src/mongocrypt-ctx-encrypt.c

```diff
@@ -1852,6 +1872,12 @@ static bool explicit_encrypt_init(mongocrypt_ctx_t *ctx, mongocrypt_binary_t *ms
         }
     }
 
+    if (ctx->opts.textopts.set && ctx->opts.index_type.set) {
```

Comment:
Recommend adding test cases in `_test_setopt_for_explicit_encrypt` to test combinations of text opts with other options. Should libmongocrypt error if text opts are set with range opts?